### ISSUE TITLE
checking if two ranges (colleactions) are equal

### ISSUE THREAD

**Issue by beling** [onqtam/doctest#74]
## Description
Please consider adding the future similar to BOOST_CHECK_EQUAL_COLLECTIONS macro from boost tests.
It is a very comfortable way to check if contents of any 2 collections are equal.
It compares two ranges (given by begins and ends - 4 parameters) and show where they mismatch.

See BOOST_CHECK_EQUAL_COLLECTIONS at
http://www.boost.org/doc/libs/1_44_0/libs/test/doc/html/utf/testing-tools/reference.html
https://stackoverflow.com/questions/3999644/how-to-compare-vectors-with-boost-test

Thank you for your great work,
Piotr Beling

**Comment by chambm** [comment]
+1. Also, unless I'm missing it, doctest is currently unable to decompose standard collections because they don't have output operators. Would it be reasonable to define output operators for the most common collections? Probably turned on/off by a preprocessor define, of course.

Here's a start:
```

namespace std
{
	template<typename T1, typename T2>
	ostream& operator<< (ostream& o, const pair<T1, T2>& p)
	{
		return (o << "( " << p.first << ", " << p.second << " )");
	}

	template<typename T>
	ostream& operator<< (ostream& o, const vector<T>& v)
	{
		o << "(";
		for (typename vector<T>::const_iterator itr = v.begin(); itr != v.end(); ++itr)
			o << " " << *itr;
		o << " )";

		return o;
	}
}
```

**Comment by onqtam** [comment]
@chambm I could provide a separate header with such overloads. But it might not be reasonable in all cases to use them - what if a vector contains a million elements? Perhaps a header with overloads might still be useful as a starting point for users to write their own versions of the operator<< overloads.

And if I add something like ```BOOST_CHECK_EQUAL_COLLECTIONS``` it should report the element indexes which are different and in that case you wouldn't want the whole containers dumped - so solving the original problem will render the need for printing collections almost obsolete.

I'll think about this, thanks.

**Comment by chambm** [comment]
Indeed. A dedicated assert type for this would render my `operator==` + `operator<<` overloads approach inferior and obsolete. :)

**Comment by cdeln** [comment]
This is a good feature proposal.
We need some examples showing how it would be used with expected outputs when the assertion fails.
